src/test/java/uk/gov/pay/ledger/util/fixture/QueueRefundEventFixture.java

```diff
@@ -0,0 +1,186 @@
+        switch (eventType) {
+            case "REFUND_CREATED_BY_SERVICE":
+                eventData = new GsonBuilder().create()
+                        .toJson(ImmutableMap.builder()
+                                .put("amount", amount)
+                                .build());
+                break;
+            case "REFUND_CREATED_BY_USER":
+                eventData = new GsonBuilder().create()
+                        .toJson(ImmutableMap.builder()
+                                .put("amount", amount)
+                                .put("refunded_by", refundedBy)
+                                .build());
+                break;
+            case "REFUND_SUBMITTED":
+                eventData = "{}";
+                break;
+            case "REFUND_SUCCEEDED" :
+                eventData = new GsonBuilder().create()
+                        .toJson(ImmutableMap.builder()
+                                .put("reference", reference)
+                                .build());
+                break;
+            default:
+                eventData = new GsonBuilder().create()
+                        .toJson(ImmutableMap.of("event_data", "event_data"));
+        }
+        return this;
+    }
+
+    @Override
+    public Event toEntity() {
+        return new Event(0L, sqsMessageId, resourceType, resourceExternalId, parentResourceExternalId, eventDate, eventType, eventData);
+    }
+
+    public String getSqsMessageId() {
+        return sqsMessageId;
+    }
+
+    public ResourceType getResourceType() {
+        return resourceType;
+    }
+
+    public String getResourceExternalId() {
+        return resourceExternalId;
+    }
+
+    public ZonedDateTime getEventDate() {
+        return eventDate;
+    }
+
+    public String getEventType() {
+        return eventType;
+    }
+
+    public String getEventData() {
+        return eventData;
+    }
+
+    @Override
+    public QueueRefundEventFixture insert(AmazonSQS sqsClient) {
+        this.withDefaultEventDataForEventType(eventType);
+        String messageBody = String.format("{" +
+                        "\"timestamp\": \"%s\"," +
+                        "\"resource_external_id\": \"%s\"," +
+                        "\"parent_resource_external_id\": \"%s\"," +
+                        "\"event_type\":\"%s\"," +
+                        "\"resource_type\": \"%s\"," +
+                        "\"event_details\": %s" +
+                        "}",
+                eventDate.toString(),
+                resourceExternalId,
+                parentResourceExternalId,
+                eventType,
+                resourceType.toString().toLowerCase(),
+                eventData
+        );
+        SendMessageResult result = sqsClient.sendMessage(SqsTestDocker.getQueueUrl("event-queue"), messageBody);
+        this.sqsMessageId = result.getMessageId();
+        return this;
+    }
+
+    public PactDslJsonBody getAsPact() {
```

Comment:
[Can be fixed in separate PR] Looks like duplicate of https://github.com/alphagov/pay-ledger/blob/8b2385ef8502053d95c8b79457babe07fa80d900/src/test/java/uk/gov/pay/ledger/util/fixture/QueueEventFixture.java#L171-L199 to me.

Reply:
Has been extracted to Util class  https://github.com/alphagov/pay-ledger/compare/8b2385ef8502053d95c8b79457babe07fa80d900..e41dffb0f31eb635371f3439403fa7a401018618#diff-27426f41fab61e239f1f824a66a05cb2R13